tfhe/src/integer/ciphertext/compressed_ciphertext_list.rs

```diff
@@ -195,4 +197,173 @@ mod tests {
 
         assert!(cks.decrypt_bool(&decompressed3));
     }
+
+    const NB_TESTS: usize = 10;
+    const NB_OPERATOR_TESTS: usize = 1;
+    #[test]
+    fn test_ciphertext_compression() {
+        let cks = ClientKey::new(PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64);
+
+        let private_compression_key =
+            cks.new_compression_private_key(COMP_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64);
+
+        let (compression_key, decompression_key) =
+            cks.new_compression_decompression_keys(&private_compression_key);
+
+        let mut rng = rand::thread_rng();
+
+        let message_modulus: u128 = cks.parameters().message_modulus().0 as u128;
+
+        let num_blocks = 32;
+        for _ in 0..NB_TESTS {
+            // Unsigned
+            let modulus = message_modulus.pow(num_blocks as u32);
+            for _ in 0..NB_OPERATOR_TESTS {
+                let nb_messages = 1 + (rng.gen::<u64>() % 6);
+                let messages = (0..nb_messages)
+                    .map(|_| rng.gen::<u128>() % modulus)
```

Comment:
Here you could generate a 64 bit integer since there are 32 blocks.

Reply:
the u128 is fine in the end I did not realize we have generic encrypt/decrypt

Reply:
Ah ok sorry

Reply:
When I do that I get an 

```thread 'integer::ciphertext::compressed_ciphertext_list::tests::test_ciphertext_compression' panicked at tfhe/src/integer/ciphertext/compressed_ciphertext_list.rs:188:30:
attempt to calculate the remainder with a divisor of zero
```

 error